pkg/kwok/metrics/evaluator_test.go

```diff
@@ -66,53 +67,150 @@ func TestNodeEvaluation(t *testing.T) {
+					Allocatable: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceCPU: resource.MustParse("90m"),
+					},
+				},
 			},
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Resources: corev1.ResourceRequirements{
-						Requests: map[corev1.ResourceName]resource.Quantity{
-							corev1.ResourceCPU: resource.MustParse("10m"),
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"cpu_usage": "10m",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceCPU: resource.MustParse("10m"),
+								},
+							},
 						},
 					},
 				},
 			},
+			expression: `( Quantity("10m") + Quantity(pod.metadata.annotations["cpu_usage"]) + node.status.allocatable["cpu"] + pod.spec.containers[0].resources.requests["cpu"] ) * 1.5 * 10`,
+			expected:   1.8,
+		},
+		{
+			name:       "parse tebibyte quantity",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1Ti")`,
+			expected:   1099511627776,
+		},
+		{
+			name:       "tebibyte multiply int",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1Ti") * 2`,
+			expected:   2199023255552,
+		},
+		{
+			name:       "tebibyte multiply float",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1Ti") * 0.5`,
+			expected:   549755813888,
+		},
+		{
+			name:       "tebibyte divide int",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1Ti") / 2`,
+			expected:   549755813888,
+		},
+		{
+			name:       "tebibyte divide float",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1Ti") / 2.0`,
+			expected:   549755813888,
+		},
+		{
+			name:       "parse millicpu quantity",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1m")`,
+			expected:   0.001,
+		},
+		{
+			name:       "millicpu multiply int",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1m") * 2`,
+			expected:   0.002,
+		},
+		{
+			name:       "millicpu multiply float",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1m") * 1.0`,
+			expected:   0.001,
+		},
+		{
+			name:       "millicpu floor to zero",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1m") * 0.5`,
+			expected:   0,
+		},
+		{
+			name:       "millicpu divide int",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("5m") / 2`,
+			expected:   0.002,
+		},
+		{
+			name:       "millicpu divide float",
+			node:       &corev1.Node{},
+			pod:        &corev1.Pod{},
+			expression: `Quantity("1m") / 0.1`,
+			expected:   0.01,
```

Comment:
Move to ./pkg/utils/cel/quantity_test.go

Reply:
Ok, I will do it.